### PR TITLE
Add initial e2e test app

### PR DESCRIPTION
**Pull request by cbracken**
Adds minimal text entry, clear test.

This is meant to serve as base upon which a comprehensive set of text
entry tests can be built. Once flutter/flutter#9383 is resolved, a set
of key-event based input tests can be built.

### CODE REVIEW

**Comment by cbracken** [comment]
Sending out as proof of concept for initial thoughts.

File: dev/devicelab/manifest.yaml [hunk 1]
```diff
@@ -150,6 +150,12 @@ tasks:
     stage: devicelab
     required_agent_capabilities: ["has-android-device"]
 
+  e2e:
+    description: >
+      Runs end-to-end Flutter tests on Android.
+    stage: devicelab
+    required_agent_capabilities: ["has-android-device"]
```
Comment (yjbanov) [inline dev/devicelab/manifest.yaml]: Are we confident that the tests will be green in the lab immediately? If not, consider marking as `flaky: true`, iterate until green, then unmarking it.

File: dev/e2e/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+name: e2e
+description: Flutter end-to-end test.
+
+dependencies:
+  flutter:
+    sdk: flutter
+dev_dependencies:
```
Comment (yjbanov) [inline dev/e2e/pubspec.yaml]: Should we be explicit that we depend on `package:test`?

File: dev/e2e/test_driver/main_test.dart [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        driver.close();
+    });
+
+    test('Test text input, clear', () async {
+      final SerializableFinder defaultTextField = find.byValueKey(keys.kDefaultTextField);
+      await driver.waitFor(defaultTextField);
+      await driver.setInputText(defaultTextField, 'Some text');
+      await driver.submitInputText(defaultTextField);
+
+      final SerializableFinder clearButton = find.byValueKey(keys.kClearButton);
```
Comment (yjbanov) [inline dev/e2e/test_driver/main_test.dart]: I miss `var` /sigh

**Review by yjbanov — APPROVED** [review]
LGTM

File: dev/e2e/pubspec.yaml (hunk 2 above)
Comment (Hixie) [inline dev/e2e/pubspec.yaml]: we should certainly be explicit that we depend on flutter_driver

File: dev/devicelab/manifest.yaml (hunk 1 above)
Comment (cbracken) [inline dev/devicelab/manifest.yaml]: Done.

File: dev/e2e/pubspec.yaml (hunk 2 above)
Comment (cbracken) [inline dev/e2e/pubspec.yaml]: Added `package:test` dep to `dev_dependencies`.
Moved `package:flutter_driver` dep from `dev_dependencies` to `dependencies`.

File: dev/e2e/test_driver/main_test.dart (hunk 3 above)
Comment (cbracken) [inline dev/e2e/test_driver/main_test.dart]: Ack :)

**Comment by Hixie** [comment]
@cbracken What's the status of this PR?

**Comment by cbracken** [comment]
I'm going to need to do #9383 first -- I'll keep the patch but drop the PR until that's ready to clean out the queue.

**Pull request closed without merging**